src/views/tutorial-view/utils/get-collection-context.ts

```diff
@@ -1,27 +1,86 @@
 import {
+  Collection as ClientCollection,
   ProductOption,
   TutorialFullCollectionCtx as ClientTutorialFullCollectionCtx,
 } from 'lib/learn-client/types'
+import { getCollection } from 'lib/learn-client/api/collection'
 import { CollectionContext } from '..'
 import { formatCollectionCard } from '../components/featured-in-collections/helpers'
 
-export function getCollectionContext(
+/**
+ * We need to get the database slug for this tutorial, which may belong in a different
+ * product directory in the filesystem. For example a tutorial with slug : `consul/get-started`
+ * may be reference in a vault collection. So we can't assume this current
+ * product context is valid for the db slug path
+ *
+ * @TODO - FOR PROD We will add a check in the content sync that
+ * ensures no two files in a collection have the same filename
+ */
+
+interface CurrentCollectionTutorial {
+  collection: {
+    filename: string
+    data: ClientCollection
+  }
+  tutorialReference: {
+    filename: string
+    dbSlug: string
+  }
+}
+
+export async function getCurrentCollectionTutorial(
   productSlug: ProductOption,
-  filename: string,
+  tutorialSlug: [string, string]
+): Promise<CurrentCollectionTutorial> {
+  /**
+   * In the db, slug structure for tutorials is {product}/{tutorial-filename}
+   * the tutorialSlug passed in is based on /{collection-name}/{tutorial-name}
+   * from the params. So we can assume `slug` index 1 is always the tutorial name
+   * */
+  const [collectionFilename, tutorialFilename] = tutorialSlug
+  const collectionDbSlug = `${productSlug}/${collectionFilename}`
+  const collection = await getCollection(collectionDbSlug)
+  /**
+   * This type is only `TutorialLite` which doesn't have the tutorial content
+   * so we only need the slug to make another request to get the full tutorial data in server.ts
+   */
+  const currentTutorial = collection.tutorials.find((t) =>
```

Comment:
Another fix here related to the cross-product tutorial issue. In order to query for the tutorial data, we need either the proper database slug (the tutorials filesystem path) or the tutorial id. In order to get the tutorial slug, we need to call for the collection and find that tutorial based on the filename.

We're planning on adding a [check in the content sync](https://app.asana.com/0/1201987349274776/1201991101510141) that prevents files with the exact same name being listed in the same collection (currently not a problem).